src/token.rs

```diff
@@ -74,3 +86,120 @@ impl Tokenizer {
+        assert_eq!(tokenizer.next().unwrap(), Token::Short('b'));
+    }
+
+    /// `-a b` should treat `b` as an arg for `-a`.
+    #[test]
+    fn one_short_flag_one_arg() {
+        let args = args_as_ref(&["binary", "-a", "b"]);
+        let mut tokenizer = Tokenizer::new_with_iterator(args.into_iter());
+        assert_eq!(tokenizer.next().unwrap(), Token::Short('a'));
+        assert_eq!(tokenizer.next_arg().unwrap(), "b");
+    }
+
+    /// `-ab` when `-a` takes an arg should be treated as `-a b`
+    #[test]
+    fn one_short_flag_one_arg_cuddled() {
+        let args = args_as_ref(&["binary", "-ab"]);
+        let mut tokenizer = Tokenizer::new_with_iterator(args.into_iter());
+        assert_eq!(tokenizer.next().unwrap(), Token::Short('a'));
+        assert_eq!(tokenizer.next_arg().unwrap(), "b");
+    }
+
+    /// `-a` when `-a` expects an arg should fail
+    #[test]
+    fn one_short_flag_missing_arg() {
+        let args = args_as_ref(&["binary", "-a"]);
+        let mut tokenizer = Tokenizer::new_with_iterator(args.into_iter());
+        assert_eq!(tokenizer.next().unwrap(), Token::Short('a'));
+        assert!(tokenizer.next_arg().is_none());
+    }
+
+    /// Long flag
+    #[test]
+    fn one_long_flag_no_args() {
+        let args = args_as_ref(&["binary", "--foo"]);
+        let mut tokenizer = Tokenizer::new_with_iterator(args.into_iter());
+        assert_eq!(tokenizer.next().unwrap(), Token::Long("foo"));
+    }
+
+    /// `--foo bar` should work
+    #[test]
+    fn one_long_flag_one_arg_space() {
+        let args = args_as_ref(&["binary", "--foo", "bar"]);
+        let mut tokenizer = Tokenizer::new_with_iterator(args.into_iter());
+        assert_eq!(tokenizer.next().unwrap(), Token::Long("foo"));
+        assert_eq!(tokenizer.next_arg().unwrap(), "bar");
+    }
+
+    /// `--foo=bar` should work
+    #[test]
+    fn one_long_flag_one_arg_equals() {
+        let args = args_as_ref(&["binary", "--foo=bar"]);
+        let mut tokenizer = Tokenizer::new_with_iterator(args.into_iter());
+        assert_eq!(tokenizer.next().unwrap(), Token::Long("foo"));
+        assert_eq!(tokenizer.next_arg().unwrap(), "bar");
+    }
+
+    /// "--" should cause everything afterwards to be treated as an argument
+    #[test]
+    fn double_dash_stops_flag_parsing() {
+        let args = args_as_ref(&["binary", "-a", "--foo", "bar", "--", "--baz", "-b", "hello"]);
+        let mut tokenizer = Tokenizer::new_with_iterator(args.into_iter());
+        assert_eq!(tokenizer.next().unwrap(), Token::Short('a'));
+        assert_eq!(tokenizer.next().unwrap(), Token::Long("foo"));
+        assert_eq!(tokenizer.next_arg().unwrap(), "bar");
+        assert_eq!(tokenizer.next_arg().unwrap(), "--");
+        // After the '--' everything should be interpreted literally
+        assert_eq!(tokenizer.next_arg().unwrap(), "--baz");
+        assert_eq!(tokenizer.next_arg().unwrap(), "-b");
+        assert_eq!(tokenizer.next_arg().unwrap(), "hello");
```

Comment:
It would be more meaningful to test:

```suggestion
        assert_eq!(tokenizer.next().unwrap(), Token::Arg(OsStr::new("bar")));
        // After the '--' everything should be interpreted literally
        assert_eq!(tokenizer.next().unwrap(), Token::Arg(OsStr::new("--baz")));
        assert_eq!(tokenizer.next().unwrap(), Token::Arg(OsStr::new("-b")));
        assert_eq!(tokenizer.next().unwrap(), Token::Arg(OsStr::new("hello")));
```

since `--` doesn't make any difference in next_arg().

Reply:
Done.